### DISCUSSION TITLE
Midi setup

### DISCUSSION THREAD

**Discussion by aefazek** [bangcorrupt/hacktribe#274] [Q&A] [answered]
Hi!

I was looking for detailed information on "Expanded MIDI implementation" as a document, but have not found one, so i would like to get some information on this.
What are those new midi features exactly ?

**Comment by bangcorrupt** [comment]
There is some information on the [MIDI](https://github.com/bangcorrupt/hacktribe/wiki/MIDI) page in the wiki.  On the `main` branch, most of the buttons and the encoders transmit NRPN messages.  Hacktribe will receive NRPN to edit FX presets or SysEx to read and write CPU RAM.  The `experimental` branch adds MIDI Soft Thru and motion sequences are sent as CC (this is still buggy).  

Editing FX presets is probably easiest using [`ht-cli`](https://github.com/bangcorrupt/hacktribe/wiki/MIDI).  NRPN and SysEx messages can be built using [ht_nrpn.py](https://github.com/bangcorrupt/hacktribe-editor-legacy/blob/ht-cli/hacktribe_editor/ht_nrpn.py) and [ht_sysex.py](https://github.com/bangcorrupt/hacktribe-editor/blob/ht-cli/hacktribe_editor/ht_sysex.py) if you want to make your own editors.  There is also an old [script for editing groove templates](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2_groove_template.py).

**Comment by aefazek** [comment]
Thanks for the reply!
To be more specific: 
Is it possible to modify e.g. program change messages sent - MSB/LSB/program/MIDI CH values ?
I am facing an issue controlling other gear as Bank 0 (LSB) sent on Pattern change, i would like to modify this default transmitted value.

Beside this advanced midi filtering options would also improve use with external gear - separate ON/OFF options for Note, CC, Program Change sent.

Thank you for this awesome project!

**Reply by bangcorrupt** [reply, marked as answer]
It's not possible yet, but in the future I hope to share source code for MIDI input and output filtering.  I'll also add configuration options for the current MIDI hacks.  This is necessary at least just to reduce the volume of MIDI data being sent.

I don't know if it helps, but I think you can get Bank Select LSB of 1 by selecting patterns higher than 127.

**Reply by aefazek** [reply]
What a surprise!
Based on the original poor midi functions and documentation I was not expecting a thing like this.
From pattern 129 - Bank LSB 1 transmitted ! YEAAH
I feel so noob this should have been a korgforums thread :facepalm:
Cheers!

**Reply by bangcorrupt** [reply]
Thank you for your support! :smile:
